lib/Dialect/ESI/cosim/MtiPliStub/DummySvDpi.cpp

```diff
@@ -0,0 +1,226 @@
+//===- DummySvDpi.cpp - Dummy stubs the svdpi.h -----------------*- C++ -*-===//
+//
+// Dummy stubs for every function in svdpi.h. This produces an MtiPli.so for
+// linking a DPI library in the absence of a simulator. Should dynamicall link
+// with the MtiPli library supplied with the simulator at runtime. This shared
+// object should not be distributed with CIRCT.
+//
+//===----------------------------------------------------------------------===//
+
+#include "external/dpi/svdpi.h"
+#undef NDEBUG
+#include <cassert>
+
+const char *svDpiVersion(void) {
+  assert(false && "Linking error: should not ever execute.");
```

Comment:
I'd recommend using `llvm_unreachable` here instead of assert(false) since the later goes away in NDEBUG configs.

Reply:
Yeah, I would have used that but I'm trying to leave the cosim code compilable without llvm. I #undef'd NDEBUG to avoid the NDEBUG config issue.